### PR TITLE
Document the various event files produced by CI

### PR DESCRIPTION
**Pull request by DennisPeriquet**

### CODE REVIEW

File: dev-guide/ci/events-files.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+* event-filter.html
+  * included in the output of a `oc adm must-gather`
+  * contains the events from all namespaces plus events that exist in the kube-apiserver at the time `oc adm must-gather` was run
+  * located in the prow job stored artifacts at `.../gather-must-gather/artifacts/` and contained in the must-gather.tar
+    file in that same location
+  * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-must-gather/artifacts/event-filter.html)
+  * Tip: wait several seconds for the html page to finish rendering so you can use the text entry boxes to search
+ 
+* events.json (same as events except as json)
```
Comment (dgoodwin) [inline dev-guide/ci/events-files.md:20]: Same as event-filter.html.

File: dev-guide/ci/events-files.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  * located in the prow job stored artifacts at `.../gather-extra/aftifacts/`
+  * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-extra/artifacts/events.json)
+ 
+ * events (same as events.json except not json)
+   * generated via `oc get events --all-namespaces` (see [source code](https://github.com/openshift/release/blob/f5017d5136a740a4186477b02bed70047ade200b/ci-operator/step-registry/gather/extra/gather-extra-commands.sh#L62))
+   * located in the prow job stored artifacts at `.../gahter-extra/artifacts/oc_cmds/events`
+   * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-extra/artifacts/oc_cmds/events)
```
Comment (dgoodwin) [inline dev-guide/ci/events-files.md:28]: Didn't know this one existed.

File: dev-guide/ci/events-files.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+* e2e-events_(date)-(number).json
+  * located in the prow job stored artifacts at `.../<test_name>/openshift-e2e-test/artifacts/junit`; there are two of them
+  * The smaller one contains ...
+  * The bigger one contains ...
+  * [Example1](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/openshift-e2e-test/artifacts/junit/e2e-events_20220321-040307.json), [Example2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/openshift-e2e-test/artifacts/junit/e2e-events_20220321-052002.json)
+
+* We talked about events related to [Devan's PR](https://github.com/openshift/origin/pull/26862) which we can document
+  more later
+  * These events are Monitor events (received by a Watcher)
```
Comment (dgoodwin) [inline dev-guide/ci/events-files.md:60]: I thought these were the e2e-events above but now I'm not so sure. :)

File: dev-guide/ci/events-files.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    
+    ./e2e-k8s-service-load-balancer-with-pdb-new-485:
+    total 88
+    -rw-rw-r-- 1 dperique dperique 86468 Dec 31  1979 events.json
+```
+
+* e2e-events_(date)-(number).json
+  * located in the prow job stored artifacts at `.../<test_name>/openshift-e2e-test/artifacts/junit`; there are two of them
+  * The smaller one contains ...
+  * The bigger one contains ...
```
Comment (dgoodwin) [inline dev-guide/ci/events-files.md:55]: My understanding was you get two files if you invoke openshift-tests twice, as we do for upgrade jobs.

**Review by dgoodwin — COMMENTED** [review]
Let's give deads a few days for the upstream stuff he's on but then ping him about this next week. We'll definitely need his clarification.

File: dev-guide/ci/events-files.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+You will see the "Artifacts" link in the upper right of a prow job (for example, see
+[this prow job](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664).
+
+This document provides details on where the files are located and what they contain.
+
+## Events files
+
+* event-filter.html
+  * included in the output of a `oc adm must-gather`
+  * contains the events from all namespaces plus events that exist in the kube-apiserver at the time `oc adm must-gather` was run
```
Comment (deads2k) [inline dev-guide/ci/events-files.md:14]: not all events, just the events in namespaces gathered by must-gather.  so namespaces not listed as relatedResources for clusteroperators will be missing.

File: dev-guide/ci/events-files.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  * Tip: wait several seconds for the html page to finish rendering so you can use the text entry boxes to search
+
+* events.json (same as `oc_cmds/events` mentioned below except as json)
+  * generated via `oc get events --all-namespaces -o json” (see [source code](https://github.com/openshift/release/blob/f5017d5136a740a4186477b02bed70047ade200b/ci-operator/step-registry/gather/extra/gather-extra-commands.sh#L61))
+  * located in the prow job stored artifacts at `.../gather-extra/aftifacts/`
+  * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-extra/artifacts/events.json)
+
+ * events (same as `gather-extra/artifacts/events.json` except not json)
+   * generated via `oc get events --all-namespaces` (see [source code](https://github.com/openshift/release/blob/f5017d5136a740a4186477b02bed70047ade200b/ci-operator/step-registry/gather/extra/gather-extra-commands.sh#L62))
```
Comment (deads2k) [inline dev-guide/ci/events-files.md:26]: this one is all the events present in the apiserver at the end.

File: dev-guide/ci/events-files.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+   * located in the prow job stored artifacts at `.../gahter-extra/artifacts/oc_cmds/events`
+   * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-extra/artifacts/oc_cmds/events)
+
+* resource-events_(date)-(number).zip
+  * located in the prow job stored artifacts at `.../<test_name>/openshift-e2e-test/artifacts/junit`; there are two of them (not sure why?)
```
Comment (deads2k) [inline dev-guide/ci/events-files.md:31]: two different openshift-tests invocations.

File: dev-guide/ci/events-files.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    ./e2e-check-for-dns-availability-1546:
+    total 68
+    -rw-rw-r-- 1 dperique dperique 67899 Dec 31  1979 events.json
+
+    ./e2e-k8s-service-load-balancer-with-pdb-new-485:
+    total 88
+    -rw-rw-r-- 1 dperique dperique 86468 Dec 31  1979 events.json
+```
+
+* e2e-events_(date)-(number).json
```
Comment (deads2k) [inline dev-guide/ci/events-files.md:52]: there are monitor events, not API events.   the monitor is part of the openshift-tests invocation.  They do not have the same info as the API events.

File: dev-guide/ci/events-files.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  * located in the prow job stored artifacts at `.../gather-extra/aftifacts/`
+  * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-extra/artifacts/events.json)
+
+ * events (same as `gather-extra/artifacts/events.json` except not json)
+   * generated via `oc get events --all-namespaces` (see [source code](https://github.com/openshift/release/blob/f5017d5136a740a4186477b02bed70047ade200b/ci-operator/step-registry/gather/extra/gather-extra-commands.sh#L62))
+   * located in the prow job stored artifacts at `.../gahter-extra/artifacts/oc_cmds/events`
+   * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-extra/artifacts/oc_cmds/events)
+
+* resource-events_(date)-(number).zip
```
Comment (deads2k) [inline dev-guide/ci/events-files.md:30]: these contain kube-apiserver events.

File: dev-guide/ci/events-files.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+* e2e-events_(date)-(number).json
+  * located in the prow job stored artifacts at `.../artifcts/<test_name>/openshift-e2e-test/artifacts/junit`; there are two of them
+  * The first one is from the initial installation
+  * The second one is from the upgrade (if initial installation fails, this one will not be present)
+  * [Example1](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/openshift-e2e-test/artifacts/junit/e2e-events_20220321-040307.json), [Example2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/openshift-e2e-test/artifacts/junit/e2e-events_20220321-052002.json)
+
+* core/events.yaml
+  * located in the must-gather.tar at `.../namespaces/(aNamespace)/core/events.yaml`; there is apparently one per namespace.
+  * ??? (I could not find where this is produced)
```
Comment (deads2k) [inline dev-guide/ci/events-files.md:60]: this is created by `oc adm inspect`, which is called by must-gather.

File: dev-guide/ci/events-files.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
+# Openshift CI produced files related to events
+
+To aid in debugging, the Openshift CI (prow) produces various files related to events.  The files
+reside in the gather-must-gather and gather-extra subdirectories in the artifacts storage area.
+You will see the "Artifacts" link in the upper right of a prow job (for example, see
+[this prow job](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664).
```
Comment (wking) [inline dev-guide/ci/events-files.md:6]: nit: we eventually garbage-collect these artifacts.  Might be worth explaining how to get to a recent run, like https://amd64.ocp.releases.ci.openshift.org/#4-stable -4.y.z-> https://amd64.ocp.releases.ci.openshift.org/releasestream/4-stable/release/4.10.12 -upgrade-> https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/1519313665751257088 ?

File: dev-guide/ci/events-files.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+This document provides details on where the files are located and what they contain.
+
+## Events files
+
+* event-filter.html
+  * included in the output of a `oc adm must-gather`
+  * contains events from namespaces gathered by `oc adm must-gather` plus events that exist in the kube-apiserver at the time `oc adm must-gather` was run.
```
Comment (wking) [inline dev-guide/ci/events-files.md:14]: how do "events that exist in the kube-apiserver" differ from the ones must-gather collected?

File: dev-guide/ci/events-files.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    Namespaces not listed as relatedResources for clusteroperators will be missing.
+  * located in the prow job stored artifacts at `.../gather-must-gather/artifacts/` and contained in the must-gather.tar
+    file in that same location
+  * [Example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/gather-must-gather/artifacts/event-filter.html)
```
Comment (wking) [inline dev-guide/ci/events-files.md:18]: before we get down to individual step content like this, it's probably worth talking about the `.../{job-name}/{run-id}/artifacts/{job-slug}/{step-name}/artifacts/` structure to get artifacts from each step.  Once we have established that pattern, we can say things like "in gather-must-gather assets" without having to steps to get there or linking to things that will be garbage collected a few months from now.

File: dev-guide/ci/events-files.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+  * Tip: wait several seconds for the html page to finish rendering so you can use the text entry boxes to search
+
+* events.json (same as `oc_cmds/events` mentioned below except as json)
+  * generated via `oc get events --all-namespaces -o json” (see [source code](https://github.com/openshift/release/blob/f5017d5136a740a4186477b02bed70047ade200b/ci-operator/step-registry/gather/extra/gather-extra-commands.sh#L61))
```
Comment (wking) [inline dev-guide/ci/events-files.md:22]: nit: unbalanced quoting aroudn the `oc ...` command here.

File: dev-guide/ci/events-files.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+  * There will be one file per invocation of `openshift-tests`
+  * Contains a watch of all events including "delete" related events
+  * These contain kube-apiserver events
+  * [Example1](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/openshift-e2e-test/artifacts/junit/resource-events_20220321-040307.zip), [Example2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1505747388038385664/artifacts/e2e-gcp-ovn-upgrade/openshift-e2e-test/artifacts/junit/resource-events_20220321-052002.zip)
+  * Here's a sample of the contents:
+
+```
+    ./default:
+    total 332
+    -rw-rw-r-- 1 dperique dperique 339013 Dec 31  1979 events.json  <-- why the ancient date?
```
Comment (wking) [inline dev-guide/ci/events-files.md:42]: that's the Unix epoc, and is just showing the timestamp property was left unset.

File: dev-guide/ci/events-files.md (hunk 12 above)
Comment (dgoodwin) [reply dev-guide/ci/events-files.md:14]: Is that true? I'd have figured events falls into a category of things must-gather would grab globally without need of related resources.

File: dev-guide/ci/events-files.md (hunk 8 above)
Comment (dgoodwin) [reply dev-guide/ci/events-files.md:53]: Would it be worth the bother of renaming these? I believe this is fairly confusing for anyone who encounters them.

Comment (deads2k) [reply dev-guide/ci/events-files.md:63]: >Would it be worth the bother of renaming these? I believe this is fairly confusing for anyone who encounters them.

I'd merge it.  I'd also merge a PR to write the raw events separate from the computed ones.

**Pull request merged**